src/LanguageServer/Impl/Telemetry/RequestTimer.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Python.Core;
+using Microsoft.Python.Core.Services;
+
+namespace Microsoft.Python.LanguageServer.Telemetry {
+    internal class RequestTimer {
+        private const int MaxEvents = 10;
+
+        private readonly ITelemetryService _telemetryService;
+
+        private readonly Dictionary<string, int> _events = new Dictionary<string, int>();
+        private readonly object _lock = new object();
+
+        public RequestTimer(ITelemetryService telemetryService) {
+            _telemetryService = telemetryService;
+        }
+
+        public Timer Time(string method) {
+            lock (_lock) {
+                _events.TryGetValue(method, out var eventNum);
+
+                if (eventNum >= MaxEvents) {
+                    return new Timer();
```

Comment:
Perhaps make static Timer.Disabled and return it, like we have in Core `Disposable.Empty`